### DISCUSSION TITLE
Pinterest: Modal not popping up

### DISCUSSION THREAD

**Discussion by caseywalker** [nss-evening-cohort-14/evening-client-side#172] [Help Me!] [answered]
Casey Walker 

Have you searched the other help tickets? Yes.
#166 Difference: Katy's modal was populating twice, my modal is not populating at all. 

Describe your issue: Modal is not populating on DOM when clicked. I have tested to make sure the div's I am targeting work, seems to be the modal is not populating. 

List any error messages that appear. None 

What have you googled? (Paste links to the things you have found from your google search - You must include at least 3 links)
https://stackoverflow.com/questions/34637451/why-is-my-modal-not-popping-up-in-the-page/34637716
https://stackoverflow.com/questions/65406733/bootstrap-modal-not-popping-up-when-clicking-button
https://stackoverflow.com/questions/28388524/bootstrap-modal-not-popping-up/28388839

What have you tried? Tell us everything you have tried.
I used the form modal from Almost Amazon, and edited it for this project. 
When that wasn't working, I copied a new modal from bootstrap. 
I ensured that the div I was targeting on the DOM would present something, it does. Seems the modal is the issue. 

Link to modal: (ignore my commented out previous modal)
https://github.com/caseywalker/ASSIGNMENT-Firebase-Pinterest/blob/update/src/javascripts/components/forms/formModal.js#L17
Link to editPinForm 
https://github.com/caseywalker/ASSIGNMENT-Firebase-Pinterest/blob/update/src/javascripts/components/forms/editPinForm.js
Link to DOM Events: 
https://github.com/caseywalker/ASSIGNMENT-Firebase-Pinterest/blob/update/src/javascripts/events/domEvents.js#L83
DOM Builder: 
https://github.com/caseywalker/ASSIGNMENT-Firebase-Pinterest/blob/update/src/javascripts/components/domBuilder.js#L8

You will see I commented out parts that I will need to update once I get the modal to populate.

**Comment by TrinityChristiana** [comment]
In order for a modal to pop up on a button click, there need to be some attributes added to the button. 

Here's the example of the Edit Book button in Almost Amazon: https://github.com/drteresavasquez/IN-CLASS-almost-amazon-template/blob/main/src/javascripts/components/books.js#L14. 

You'd just need to update this line here: https://github.com/caseywalker/ASSIGNMENT-Firebase-Pinterest/blob/update/src/javascripts/components/pins.js#L14

**Comment by caseywalker** [comment, marked as answer]
SOLUTION: Buttons that will be targeted for click events with modals need these attributes: ```data-toggle="modal" data-target="#formModal" ```
